tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/object/ObjectInBaloTest.java

```diff
@@ -571,7 +571,9 @@ public void testObjectReferingTypeFromBaloNegative() {
         int i = 0;
         BAssertUtil.validateError(result, i++, "undefined field 'name' in object 'Manager1'", 25, 13);
         BAssertUtil.validateError(result, i++, "undefined field 'age' in object 'Manager1'", 26, 13);
-        BAssertUtil.validateError(result, i, "incompatible types: 'foo:Manager1' is not an abstract object", 38, 6);
+        BAssertUtil.validateError(result, i,
+                                  "incompatible type reference 'foo:Manager1': a referenced object cannot have " +
+                                          "non-public fields or methods", 38, 6);
     }
 
```

Comment:
We need to add a balo test case for this feature.

Reply:
Added : 8e8601896a6850ade29a643e2e92a1153f220a2e